<launch>

    <!--  reads data from DWM1001 by serial port and publishes in 'localization' chatter -->
   <node pkg="Augbot" name="reader" type="localization_improved_publisher"/>

    <!--  subscribes 'localization' chatter and publishes tagFull in mqtt  -->
    <node name="mqttBridgeTagFull" pkg="Augbot" type="mqttBridge.py" output="screen">
      <param name="subset" value="tagFull" /> 
      <param name="toDo" value="mqttPublish" /> 
      <rosparam command="load" file="$(find Augbot)/config/mqttBridge_params.yaml" />
    </node>

    <!--   subscribes 'localization' chatter, transform data and publishes 'least_squares' chatter -->
    <node pkg="Augbot" name="least_squares" type="least_squares.py"/>

      <!--  subscribes 'least_squares' chatter and publishes position in mqtt  -->
    <node name="mqttBridgeLS" pkg="Augbot" type="mqttBridge.py" output="screen">
      <param name="subset" value="leastSquares" /> 
      <param name="toDo" value="mqttPublish" /> 
      <rosparam command="load" file="$(find Augbot)/config/mqttBridge_params.yaml" />
    </node>

</launch>

